### PR TITLE
WIP: Fix `error_reporting()` with PHP >= 8

### PR DESCRIPTION
**Pull request by drupol**
This PR:

- [x] Needs #1377 in order to work

Relates to #1375

### CODE REVIEW

File: spec/PhpSpec/Runner/Maintainer/ErrorMaintainerSpec.php [hunk 1]
```diff
@@ -30,10 +30,4 @@ function it_return_true_when_recoverable_level_and_message_match()
         $msg = 'Argument 1 passed to ' . self::class . '::test() must be an instance of string, string given';
         $this->errorHandler(E_RECOVERABLE_ERROR, $msg, 'file', 1)->shouldBe(true);
     }
-
-    function it_throws_error_exception_when_message_not_match()
```
Comment (ciaranmcnulty) [inline spec/PhpSpec/Runner/Maintainer/ErrorMaintainerSpec.php:34]: Why is this deleted?

Comment (drupol) [reply spec/PhpSpec/Runner/Maintainer/ErrorMaintainerSpec.php:34]: I think this is not needed anymore in PHP 8.

Could be nice to have confirmation from @alexandrugg or @kelunik ?

Comment (AlexandruGG) [reply spec/PhpSpec/Runner/Maintainer/ErrorMaintainerSpec.php:34]: are you saying it's not possible to test the exception path anymore?

Comment (AlexandruGG) [reply spec/PhpSpec/Runner/Maintainer/ErrorMaintainerSpec.php:34]: I think we need a couple of extra tests to show the behaviour we want from this actually

Comment (drupol) [reply spec/PhpSpec/Runner/Maintainer/ErrorMaintainerSpec.php:34]: At least in PHP 8, yes.

Comment (AlexandruGG) [reply spec/PhpSpec/Runner/Maintainer/ErrorMaintainerSpec.php:34]: if I test that with `32767 E_ALL` it does throw the exception in both php 7 and 8:
```PHP
function it_throws_error_exception_when_message_not_match()
    {
        $this->shouldThrow(ExampleException\ErrorException::class)
             ->during('errorHandler', [E_ALL, 'error message', 'file', 1]);
    }
```


this being said I can't claim to fully understand this code change 😅 .

Would it be useful to have a test where the `@` [operator](https://www.php.net/manual/en/language.operators.errorcontrol.php) is used to suppress something that would've caused the exception to be thrown in PHP 8, and now with this change it's no longer thrown? Because that's kind of what we're looking to accomplish with this

Comment (kelunik) [reply spec/PhpSpec/Runner/Maintainer/ErrorMaintainerSpec.php:34]: I don't see a reason to delete any test.

Comment (drupol) [reply spec/PhpSpec/Runner/Maintainer/ErrorMaintainerSpec.php:34]: I deleted the test, maybe I was wrong.

Anyway, despite that, I don't understand the meaning of this test.

If anyone could help, I would be very grateful :)

Comment (AlexandruGG) [reply spec/PhpSpec/Runner/Maintainer/ErrorMaintainerSpec.php:34]: I'll try some stuff so I can better wrap my head around this

**Comment by ciaranmcnulty** [comment]
It'd be good to get more coverage of different error scenarios - maybe in the behat features

File: spec/PhpSpec/Runner/Maintainer/ErrorMaintainerSpec.php (hunk 1 above)
Comment (AlexandruGG) [reply spec/PhpSpec/Runner/Maintainer/ErrorMaintainerSpec.php:34]: I opened a separate PR here cause I didn't think I could push to yours: https://github.com/phpspec/phpspec/pull/1378

**Comment by drupol** [comment]
Closing in favor of #1378
